testsuite/classlibrary/TestUnitTest.sc

```diff
@@ -1,13 +1,34 @@
 TestUnitTest : UnitTest {
 
-	var someVar;
+	var someVar, setUpHappened = false;
+	classvar setUpHappened = false;
+	classvar tearDownHappened = false;
+
+	*setUpClass {
+		setUpHappened = true;
+	}
+
+	*tearDownClass {
+		tearDownHappened = true;
+	}
 
 	setUp {
-		someVar = \setUp;
+		setUpHappened = true;
+	}
+
+	test_setUpClass_in_use {
+		this.class.use {
```

Comment:
should not need to use `use` directly to get the effects of setUpClass/tearDownClass

Reply:
This test is intended to test whether setUp is called in `use`, so we need to call it.

Reply:
but, `classSetUpHappened` should be set before `test_setUpClass_in_use` is even invoked. so if you are going to test `use` you should make sure to set `classSetUpHappened` to false beforehand. similarly, `test_tearDownClass_in_use` should set `classTearDownHappened` to false before running `this.class.use`. i would also recommend changing the test method names to emphasize that they are testing `use` and not `tearDownClass`/`setUpClass`. finally, i would also suggest a test for `classSetUpHappened` that just checks the value is true. we can't really test `classTearDownHappened` without something a little more elaborate, i think.

Reply:
note that i am separating the desire to test `use` from the desire to test that test methods themselves are always run inside `use`. i think we should have some simple test cases for both.